test/NVMLAcceleratorTopoTest.cpp

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2015, 2016, 2017, 2018, 2019, 2020, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <unistd.h>
+#include <limits.h>
+
+#include <fstream>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "config.h"
+#include "Helper.hpp"
+#include "Exception.hpp"
+#include "MockNVMLDevicePool.hpp"
+#include "NVMLAcceleratorTopo.hpp"
+
+using geopm::NVMLAcceleratorTopo;
+using geopm::Exception;
+using testing::Return;
+
+class NVMLAcceleratorTopoTest : public :: testing :: Test
+{
+    protected:
+        void SetUp();
+        void TearDown();
+
+        std::shared_ptr<MockNVMLDevicePool> m_device_pool;
+};
+
+void NVMLAcceleratorTopoTest::SetUp()
+{
+    m_device_pool = std::make_shared<MockNVMLDevicePool>();
+}
+
+void NVMLAcceleratorTopoTest::TearDown()
+{
+}
+
+//Test case: Mock num_accelerator = 0 so we hit the appropriate warning and throw on affinitization requests.
+TEST_F(NVMLAcceleratorTopoTest, no_gpu_config)
+{
+    int num_accelerator = 0;
+    int num_cpu = 40;
+
+    EXPECT_CALL(*m_device_pool, num_accelerator()).WillOnce(Return(num_accelerator));
+
+    NVMLAcceleratorTopo topo(*m_device_pool, num_cpu);
+    EXPECT_EQ(0, topo.num_accelerator());
+
+    EXPECT_THROW(topo.ideal_cpu_affinitization(0), geopm::Exception);
```

Comment:
If you want to get fancy, we have ``` GEOPM_EXPECT_THROW_MESSAGE``` which allows you to look for a particular substring in the exception text.  It's more specific than just catching any geopm::Exception that occurs here.

There's a ton of examples of usage in the test/ dir.